src/pkg/golang/build.go

```diff
@@ -204,19 +232,107 @@ func (c Environ) GoCmd(gocmd string, args ...string) *exec.Cmd {
 	return cmd
 }
 
-// Version returns the Go version string that runtime.Version would return for
-// the Go compiler in this environ.
-func (c Environ) Version() (string, error) {
+// resolve absolute path of go-compiler option from PATH
+func (c *Environ) resolveCompiler() error {
+	if c.Compiler.Path != "" {
+		fname, err := exec.LookPath(string(c.Compiler.Path))
+		if err == nil {
+			fname, err = filepath.Abs(fname)
+		}
+		if err != nil {
+			return fmt.Errorf("build: %v", err)
+		}
+		c.Compiler.Path = fname
+	}
+	return nil
+}
+
+// runs GoCmd("version") and parse/caches output, to environ.Compiler
+func (c *Environ) InitCompiler() error {
+
+	if c.Compiler.IsInit {
+		return nil
+	}
+
+	c.resolveCompiler()
+
 	cmd := c.GoCmd("version")
 	v, err := cmd.CombinedOutput()
 	if err != nil {
-		return "", err
+		return err
 	}
+
+	efmt := "go-compiler 'version' output unrecognized: %v"
```

Comment:
here's what I'd do for errors.

```
var ErrNoGoVersion = fmt.Errorf("go-compiler version output unrecognized")

...

return fmt.Errorf("%w: %s", ErrNoGoVersion, string(v))
```

This allows using errors.Is with `ErrNoGoVersion` and keeping the error message the same for the two efmt cases